engine/engine.go

```diff
@@ -688,6 +723,28 @@ func recoverEngine(logger log.Logger, expr parser.Expr, errp *error) {
 	}
 }
 
+func analyze(w io.Writer, o model.ObservableVectorOperator, indent, indentNext string) {
```

Comment:
Nice! Got an example of an output with this?

Reply:
![Screenshot from 2023-07-07 23-06-02](https://github.com/thanos-io/promql-engine/assets/99465982/34518315-c36b-46ac-91d9-35cacbdea182)
This is what i get on local testing the analyze function

Reply:
Why not add a code test? 🙂